app/src/main/java/com/example/androidthings/testapp/controleur/Temperature.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.androidthings.testapp.controleur
+
+import android.util.Log
+import com.example.androidthings.testapp.singleton.SerialSingleton
+import org.json.JSONObject
+import org.restlet.data.MediaType
+import org.restlet.ext.json.JsonRepresentation
+import org.restlet.representation.Representation
+import org.restlet.representation.StringRepresentation
+import org.restlet.resource.Get
+import org.restlet.resource.Post
+import org.restlet.resource.ServerResource
+import utils.TAG
+
+/** Controleur to get temperature from android things Device**/
+class Temperature: ServerResource() {
+
+    /** if post :
+     * @param json object
+     * call getTemp function
+     * @return json with temperature
+     **/
+    @Get("")
+    fun getState(entity: JsonRepresentation): Representation {
+        var result = JSONObject()
+        var get = "404 error"
+
+        Log.i(TAG, "@Get : $result")
```

Comment:
you are loging the result before the affection (line 32). why ?

Reply:
Just creating an empty JSONObject which will be complete later. Thought that in kotlin it is better to initiate variables rather than overusing "lateinit" or allowing variable to be null. Once the JSON object is created i put the "temp" val inside after.

Reply:
For a HTTP GET, the service should create the json by itself